src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -62,13 +64,43 @@ def resolve_requirements(self, req_libs, local_dist_targets=None):
           # Handle locally-built python distribution dependencies.
           built_dists = self.context.products.get_data(BuildLocalPythonDistributions.PYTHON_DISTS)
           if built_dists:
-            req_libs = inject_synthetic_dist_requirements(self.context.build_graph,
-                                                          built_dists,
-                                                          ':'.join(2 * [target_set_id])) + req_libs
-          self._build_requirements_pex(interpreter, safe_path, req_libs)
+            req_libs = inject_synthetic_dist_requirements(
+              self.context.build_graph, built_dists, ':'.join(2 * [target_set_id])
+            ) + req_libs
+          builder = PEXBuilder(path=safe_path, interpreter=interpreter, copy=True)
+          dump_requirement_libs(builder, interpreter, req_libs, self.context.log)
+          builder.freeze()
     return PEX(path, interpreter=interpreter)
 
-  def _build_requirements_pex(self, interpreter, path, req_libs):
-    builder = PEXBuilder(path=path, interpreter=interpreter, copy=True)
-    dump_requirements(builder, interpreter, req_libs, self.context.log)
-    builder.freeze()
+  def resolve_requirement_strings(self, interpreter, requirement_strings):
```

Comment:
I think the name of this method could be more descriptive, like `build_pex_from_requirement_strings`, but this is just my opinion and the current name is in line with the previous name, so it also makes sense to leave as is. Just a suggestion.